src/Cli/dotnet/ProjectLocator.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Execution;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Cli
+{
+    abstract class ProjectLocator
```

Comment:
Good question. The code that `dotnet run` is doing is very similar. So this allows us to reduce some of the duplication if we want to do that in future. Didn't do that here because would like to keep this as 'simple' as possible